### PR TITLE
Log block arrival time

### PR DESCRIPTION
**Pull request by potuz**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**
Logs block arrival time through pubsub, thus getting a complete log of block arrival time and total block processing time.

### CODE REVIEW

File: beacon-chain/sync/validate_beacon_blocks.go [hunk 1]
```diff
@@ -140,6 +144,16 @@ func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, ms
 	// Record attribute of valid block.
 	span.AddAttributes(trace.Int64Attribute("slotInEpoch", int64(blk.Block.Slot%params.BeaconConfig().SlotsPerEpoch)))
 	msg.ValidatorData = blk // Used in downstream subscriber
+	//Log the arrival time of the accepted block
```
Comment (terencechain) [inline beacon-chain/sync/validate_beacon_blocks.go:147]:
```suggestion

	// Log the arrival time of the accepted block
```

**Review by terencechain — COMMENTED** [review]
I'm not convinced that we really need this log.

Why can't we use tracing and observe `validateBeaconBlockPubSub`?

**Comment by potuz** [comment]
> I'm not convinced that we really need this log.
> 
> Why can't we use tracing and observe `validateBeaconBlockPubSub`?

I can't really afford to use trace logs continuously, and this kind of debug, when used, is typically for a bad vote/missed attestation that may have happened hours before. As you said in a previous PR, this only adds a few bytes per successful block sent to the blockchain and only the first time it's seen.

**Comment by terencechain** [comment]
> > I'm not convinced that we really need this log.
> > Why can't we use tracing and observe `validateBeaconBlockPubSub`?
> 
> I can't really afford to use trace logs continuously, and this kind of debug, when used, is typically for a bad vote/missed attestation that may have happened hours before. As you said in a previous PR, this only adds a few bytes per successful block sent to the blockchain and only the first time it's seen.

Sorry I meant jaeger tracing

**Comment by rauljordan** [comment]
cc @potuz

**Comment by potuz** [comment]
> cc @potuz

If this was because of Terence's comment I have no position. It's true but then that probably applies to other timings that are logged currently in debug. I am not willing to go through installing and learning jaegger so I'll implement this on my own node just in case, but I understand that it becomes not necessary if other profiling tools are used

File: beacon-chain/sync/validate_beacon_blocks.go [hunk 2]
```diff
@@ -19,13 +19,16 @@ import (
...
 	"github.com/prysmaticlabs/prysm/shared/traceutil"
+	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
 )
 
 // validateBeaconBlockPubSub checks that the incoming block has a valid BLS signature.
 // Blocks that have already been seen are ignored. If the BLS signature is any valid signature,
 // this method rebroadcasts the message.
 func (s *Service) validateBeaconBlockPubSub(ctx context.Context, pid peer.ID, msg *pubsub.Message) pubsub.ValidationResult {
+	// Record the timestamp to log
```
Comment (terencechain) [inline beacon-chain/sync/validate_beacon_blocks.go:30]:
```suggestion
```

**Review by terencechain — APPROVED** [review]

**Pull request merged**